source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -3718,7 +3720,8 @@ void Graph::addNodePopup(bool cursor)
                 // Allow spaces to be used to search for node names
                 std::replace(subs.begin(), subs.end(), ' ', '_');
 
-                if (str.find(subs) != std::string::npos)
+                if ((subs.size() == 0 || str.find(subs) != std::string::npos) &&
+                    (_menuFilterType == mx::EMPTY_STRING || node.getType() == _menuFilterType))
```

Comment:
Is this the correct filtering logic? It does filter properly however I'm not sure this is the right way to filter. Given I'm pulling a pin from a `color3` this would:
1. Get the `color3` type from the `_pinFilterType`
2. Assign that type to `_menuFilterType`
3. Filter nodes by that type

This doesn't feel quite right to me since these types seem to be output types rather than input types. 

It's unclear to me how to address this as I can imagine, using the same example, you could have many nodes that would have `color3` inputs, but what would be the desired behavior in how to link them up, I imagine the first input of the pin type could be matched. 

I'll continue down this path to try to filter based on input pin types rather than output types